
---
title: "ping-pong_zscores"
author: "Filip Wierzbicki"
date: "6/14/2023"
output: rmarkdown::github_document
---
  
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{bash,eval=FALSE}
for i in *.sort.bam; do n=${i%.sort.bam}; samtools view $i | python2.7 /Volumes/Temp3/filip/programs/roberts_scripts/te-tools-code/piRNA/ping-pong-signature.py --sam - --max-mm 2 --sample-id ${n} > /Volumes/Temp3/filip/trap_model/pps/pps/${n}.pps ; done

cat *.pps|awk '$3=="s"' > forR/summary.pps.forR
```

```{r, eval=TRUE}
library(ggplot2)
library(tidyverse)
library(ggpubr)

h<-read.table("/Volumes/Temp3/filip/trap_model/pps/pps/forR/summary.pps.forR")
names(h)<-c("sid","fam","strand","pos","pps")
h$sid<-as.factor(h$sid)

# h excluding 10 
h$key<-paste0(h$sid,"+",h$fam)
het<-subset(h,pos!=10)
hin<-subset(h,pos==10)

het<-het %>%group_by(sid,fam)%>% dplyr::summarise(mean=mean(pps),sd=sqrt(var(pps)),key=key[1])
m<-merge(x=het,y=hin, by = "key")
m$zscore<-(m$pps-m$mean)/m$sd
m$zround<-round(m$zscore,2)
m$ppsround<-round(m$pps,2)
m$prob<-pnorm(m$zscore, mean = 0, sd = 1, lower.tail = FALSE)
m$probround<-signif(m$prob, digits=3)
m$sid<-m$sid.x
m$fam<-m$fam.x


t<-subset(m,select=c("sid","fam","zscore","zround","key"))

#######
###For population frequency Info based on Kofler et al. 2015 PLOS Genetics
info1<-read.table("/Users/filipwierzbicki/Desktop/evolution_cluster/temp/TEfamInfo_correct")
names(info1)<-c("name","fam","order","AF","popins")

###exclude somatically regulated TEs based on Malone et al. 2009 Cell
info1<-subset(info1,name!="gypsy10"&name!="gypsy"&name!="ZAM"&name!="gtwin"&name!="gypsy5"&name!="Tabor")

info<-subset(info1,select=c("fam","AF","name"))
info$AF<-round(info$AF,digits = 1)


ht<-left_join(t,info,by="fam")

#including AF threshold
ht<-subset(ht,AF!="NA")##remove missing AFs
ht<-subset(ht,AF<=0.2)

#ht<-subset(ht,select=c("name","Canton-S","DGRP-732","Iso1","Oregon-R","Pi2"))

############
b1<-read.table("/Volumes/Temp3/filip/trap_model/whole-genome/repeatmasker/Canton-S.fasta.out",fill=TRUE)
b1$strain<-c("Canton-S")
b1<-subset(b1, select = -c(V16))
b2<-read.table("/Volumes/Temp3/filip/trap_model/whole-genome/repeatmasker/DGRP-732.fasta.out",fill=TRUE)
b2$strain<-c("DGRP-732")
#b2<-subset(b2, select = -c(V16))
b3<-read.table("/Volumes/Temp3/filip/trap_model/whole-genome/repeatmasker/Iso1.fasta.out",fill=TRUE)
b3$strain<-c("Iso1")
#b3<-subset(b3, select = -c(V16))
b4<-read.table("/Volumes/Temp3/filip/trap_model/whole-genome/repeatmasker/Oregon-R.fasta.out",fill=TRUE)
b4$strain<-c("Oregon-R")
b4<-subset(b4, select = -c(V16))
b5<-read.table("/Volumes/Temp3/filip/trap_model/whole-genome/repeatmasker/Pi2.fasta.out",fill=TRUE)
b5$strain<-c("Pi2")
b5<-subset(b5, select = -c(V16))
b<-rbind(b1,b2,b3,b4,b5)
b$div<-as.numeric(as.character(b$V2))
b<-subset(b,div<=10.0)
b$start<-as.numeric(as.character(b$V6))
b$end<-as.numeric(as.character(b$V7))
b$len<-b$end-b$start+1
b<-subset(b,len>=100)
b$key<-paste0(b$strain,"+",b$V10)

for (sid in unique(b$key)) { 
  i <- b$key == sid
  a = nrow(subset(b,key==sid))
  b$sum[i] = a
}

TE<-subset(b,select=c("key","sum"))
TE<-unique(TE)
#TE<-subset(TE,sum>4)
######

htTE<-inner_join(ht,TE,by="key")
t<-subset(htTE,select=c("sid","name","zround")) #"fam"
#######

tx<-t %>% 
  group_by(sid) %>% 
  mutate(id = name) %>%  ##"fam"
  spread(sid, zround) %>%
  select(-id)

td<-as.data.frame(tx)
knitr::kable(td)

write.table(td,file="/Volumes/Temp3/filip/trap_model/pps/results/zscores.txt",quote = FALSE, row.names = FALSE)

############
##cusco+TAS
t1<-read.table("/Users/filipwierzbicki/Desktop/trap_model/analysis/abu/cusco_tas/gapped_combined-distinct/Canton-S_gapped_cusco_tas_summary.forR")
t2<-read.table("/Users/filipwierzbicki/Desktop/trap_model/analysis/abu/cusco_tas/gapped_combined-distinct/DGRP-732_gapped_cusco_tas_summary.forR")
t3<-read.table("/Users/filipwierzbicki/Desktop/trap_model/analysis/abu/cusco_tas/gapped_combined-distinct/Iso1_gapped_cusco_tas_summary.forR")
t4<-read.table("/Users/filipwierzbicki/Desktop/trap_model/analysis/abu/cusco_tas/gapped_combined-distinct/Oregon-R_gapped_cusco_tas_summary.forR")
t5<-read.table("/Users/filipwierzbicki/Desktop/trap_model/analysis/abu/cusco_tas/gapped_combined-distinct/Pi2_gapped_cusco_tas_summary.forR")


tq<-rbind(t1,t2,t3,t4,t5)
names(tq)<-c("count","id","fam","region")

ht<-tq
ht$id2<-paste(ht$id,ht$fam,sep="+")

cl<-subset(ht,region=="cluster")
cl<-subset(cl,select=c("count","id2"))

rest<-subset(ht,region!="cluster")
rest<-subset(rest,select=c("count","id2"))
for (sid in unique(rest$id2)) { 
  i <- rest$id2 == sid
  a = sum(rest$count[i])
  rest$sum[i] = a
}
rest<-subset(rest,select=c("sum","id2"))
names(rest)<-c("count","id2")
rest<-unique(rest)

cr<-full_join(cl,rest,by="id2")
names(cr)<-c("cluster","id2","noncluster")

cr$id<-gsub("\\+.*","",cr$id2)
cr$fam<-gsub(".*\\+","",cr$id2)
ht<-cr
ht<-left_join(ht,info,by="fam")

#including AF threshold
ht<-subset(ht,AF!="NA")##remove missing AFs
ht<-subset(ht,AF<=0.2)

ht[is.na(ht)] <- 0

missing<-subset(ht,cluster==0)

for (row in 1:nrow(htTE)) { 
  if (htTE$key[row] %in% missing$id2){
    htTE$missing[row]<-c("without cluster insertion")
  } else {
    htTE$missing[row]<-c("with cluster insertion")
  }
}

#compare_means(zscore ~ missing, data = htTE)
p <- ggboxplot(htTE, x = "missing", y = "zscore",add = "jitter")+xlab("missing piRNA cluster insertions")+ylab("ping-pong Z-score")+geom_hline(yintercept = 1.65,color="red")+theme(axis.title.x=element_blank())
p + stat_compare_means(label.x = 2, label.y = 90)
ggsave("/Volumes/Temp3/filip/trap_model/pps/results/figures/zscores.pdf",width=8,height=7)
ggsave("/Volumes/Temp3/filip/trap_model/pps/results/figures/zscores.png",width=8,height=7)

x<-subset(htTE,missing=="with cluster insertion")
y<-subset(htTE,missing=="without cluster insertion")
wilcox.test(x$zscore,y$zscore)

```

